File: controller/site.go
package controller

import (
	"net/http"
	"strconv"
	"strings"

	"github.com/gin-gonic/gin"
	"github.com/jayson2hu/image-show/model"
	"github.com/jayson2hu/image-show/service"
)

func SiteConfig(c *gin.Context) {
	c.JSON(http.StatusOK, gin.H{
		"site_title":         model.GetSettingValue("site_title", adminSettingDefaults["site_title"]),
		"site_about":         model.GetSettingValue("site_about", adminSettingDefaults["site_about"]),
		"seo_title":          model.GetSettingValue("seo_title", adminSettingDefaults["seo_title"]),
		"seo_keywords":       model.GetSettingValue("seo_keywords", adminSettingDefaults["seo_keywords"]),
		"seo_description":    model.GetSettingValue("seo_description", adminSettingDefaults["seo_description"]),
		"register_enabled":   model.RegisterEnabled(),
		"credit_costs":       service.CreditCostsByRatio(),
		"greeting_text":      model.GetSettingValue("greeting_text", adminSettingDefaults["greeting_text"]),
		"guest_free_credits": guestFreeCredits(),
	})
}

func guestFreeCredits() int {
	value := strings.TrimSpace(model.GetSettingValue("guest_free_credits", adminSettingDefaults["guest_free_credits"]))
	credits, err := strconv.Atoi(value)
	if err != nil || credits < 0 {
		return 5
	}
	return credits
}

File: web/src/api/site.ts
import api from '@/api'

export interface SiteConfig {
  site_title: string
  site_about: string
  seo_title: string
  seo_keywords: string
  seo_description: string
  register_enabled?: boolean
  credit_costs?: CreditCosts
  greeting_text?: string
  guest_free_credits?: number
  guest_generation_limit?: number
  guest_layered_generation_limit?: number
  user_generation_limit?: number
  user_layered_generation_limit?: number
}

export interface CreditCosts {
  square: number
  portrait: number
  story: number
  landscape: number
  widescreen: number
}

export function fetchSiteConfig() {
  return api.get<SiteConfig>('/site/config')
}
